internal/flags/apiflags/flags_test.go

```diff
@@ -0,0 +1,295 @@
+package apiflags
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/spf13/pflag"
+	"gotest.tools/assert"
+)
+
+func TestAddTo(t *testing.T) {
+	testCases := []struct {
+		name     string
+		apiFlags APIFlags
+		validate func(APIFlags, *pflag.FlagSet) error
+	}{
+		{
+			// Populate FlagSet
+			name: "Populate FlagSet",
+			apiFlags: APIFlags{
+				APIVersion:       "app.example.com/v1alpha1",
+				Kind:             "AppService",
+				SkipGeneration:   false,
+				CrdVersion:       "v1",
+				HelmChartRef:     "stable/app",
+				HelmChartVersion: "1.0.0",
+				HelmChartRepo:    "https://charts.mycompany.com/",
+			},
+			validate: func(apiFlags APIFlags, flagSet *pflag.FlagSet) error {
+				val, err := flagSet.GetString("api-version")
+				if err != nil {
+					return err
+				}
+				if apiFlags.APIVersion != val {
+					return fmt.Errorf("APIVersion does not match")
+				}
```

Comment:
Nit: this if for a follow on PR. We should probably create a local utility method in this test file to simplify this so the test is easier to read. I will create an issue after this PR merges.